### ISSUE TITLE
Cannot change resourceContext between stories

### ISSUE THREAD

**Issue by theKashey** [atlassian-labs/react-resource-router#76]
Discovered in Storybook environment, when was not able to reset `resourceContext` between different stories.

Expectation: mounting/unmounting Router and especially MemoryRouter are not by previously used components.

**Comment by albertogasparin** [comment]
Yeah, currently only `UniversalRouter` sets up a `ResourceContainer` that can be set to local so works with storybooks and tests. Should be addressed by the cleanup of the multiple routers we export, where `Router` can just handle this stuff properly

**Comment by theKashey** [comment]
In my case, I was able to resolve the current situation by mocking `useResource` and calling `.getData` on the given object directly, which sound like a correct solution for storybooks, where "page" and the corresponding "route" does not exist and resource in normal terms should not be fetched.

**Comment by theKashey** [comment]
Still waiting for #145

**Comment by tpuric** [comment]
Hi @theKashey , I managed to create a workaround for this without needing to override any imports.

You can create a decorator utilising `key` (for router uniqueness to ensure unmounts are handled appropriately) and forcefully destroy any existing stores when mounting another Router.

```js
export const withRouterDecoratorFn = ({ resources }: RouterDecoratorProps = {}): DecoratorFn =>
  function RouterDecorator(Story) {
    defaultRegistry.stores.clear();

    return (
      <Router
        key={key}
        routes={routes}
        history={history}
        resourceContext={routerContext}
      >
        <Story />
      </Router>
    );
  };
```

**Comment by theKashey** [comment]
Please remove `defaultRegistry.stores.clear();` there is a better place to put such destructive side effect into

As for the `Router` - this is the current recommendation.
